File: longest-common-prefix/longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        StringBuilder str = new StringBuilder();

        for (int i = 0; i < strs[0].length(); i++) {
            char a = strs[0].charAt(i);
            boolean flag = false;
            for (String s : strs) {
                if (i >= s.length()) {
                    flag = false;
                    break;
                }

                char c = s.charAt(i);
                if (c != a) {
                    flag = false;
                    break;
                }
                flag = true;
            }
            if (flag) {
                str.append(a);
            } else {
                break;
            }
        }

        return str.toString();
    }
}

File: longest-common-prefix/longest-common-prefix.kt
class Solution {
    fun longestCommonPrefix(strs: Array<String>): String {
        var overlapStr = ""
        val strsArray = strs.map { str ->
            if(str == "") {
                return overlapStr
            }
            str.toCharArray()
        }

        if(strsArray.size == 1) {
            return strs[0]
        }

        for (i in strsArray[0].indices) {
            var flag = true
            var str = ""
            for(j in 1 until strsArray.size) {
                if(i == strsArray[j].size) {
                    flag = false
                    break
                }
                if(strsArray[0][i] == strsArray[j][i]) {
                    str = strsArray[0][i].toString()
                } else {
                    flag = false
                    break
                }
            }

            if (flag) {
                overlapStr += str
            } else {
                break
            }
        }

        return overlapStr
    }
}